data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/config/MetricsConfig.java

```diff
@@ -53,7 +52,7 @@ public YAMLFactory yamlFactory() {
 
     @Bean
```

Comment:
We should name this bean.

```
@Bean(name = "metricsObjectMapper")
public ObjectMapper objectMapper(final YAMLFactory yamlFactory) {
```

And then whatever metrics-related class is using this should use the qualified form.

```
methodName(@Qualifier("metricsObjectMapper") final ObjectMapper objectMapper)
```

I'm honestly not sure what metrics code uses this.

Reply:
This is also used in `DataPrepperAppConfiguration`, even I couldn't find where and how it's being used. I don't want to rename this to "metricsObjectMapper" as it's being used to Deserialize metricsObjectMapper as well.

 I think there is a bigger issue with how these beans are created and used by other Configuration classes.

Reply:
I'm not sure this is used for metrics at all. I don't see it being used by any of these metrics beans. And it is configured for YAML. I think our metrics are output in JSON.

Reply:
I suggest that we just remove the `ObjectMapper` from the metrics config entirely.

Reply:
Or, let's qualify the peer-forwarder `ObjectMapper` for now. That should be the safest option. We should circle back to see how we can improve this in the future.